### PR TITLE
Add connection between jclklib proxy and ptp4l

### PR DESCRIPTION
**Pull request by JunAnnLaiIntel**
This PR is to add the connection between jclklib proxy and ptp4l (uds socket) by using ptpmgmt library.
For testing purpose, there is printing out the msg "received time sync data" when both ptp4l server and client are synced, to make sure the connection is established.

### CODE REVIEW

**Review by AzuraTarmiziIntel — COMMENTED** [review]
Hey peter, 
can we have a doxygene for every function.

**Comment by AzuraTarmiziIntel** [comment]
one more thing.  this is WIP . but is it enough for a incremental checkin ?

File: jclklib/proxy/thread.hpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	bool			 initialized;
+	unsigned		  valid_mask;
+	bool			peer_present;
+	unsigned		      offset;
+	bool			  gm_present;
+	bool			servo_locked;
+};
+
+int connect_ptp4l( ptp4l_handle_t *phandle, struct epoll_event epd_event, state_update_t update, void *ctx);
+int disconnect_ptp4l( ptp4l_handle_t handle);
```
Comment (AzuraTarmiziIntel) [inline jclklib/proxy/thread.hpp:134]: where are we calling the disconnect_ptp4l  ?
Or not yet ?

File: jclklib/proxy/thread.hpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ *    0 : success
+ *
+ * On failure, handle is invalid, NULL path argument uses default
+ */
+int handle_connect( jcl_handle_t *handle, struct epoll_event epd_event );
+
+void *loop(void *arg);
+bool send_subscription(struct jcl_handle **handle);
+
+
```
Comment (AzuraTarmiziIntel) [inline jclklib/proxy/thread.hpp:85]: extra spacing

File: jclklib/proxy/thread.hpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+#define jcl_servo_locked_flag			  ( 0x1 << jcl_gm_present)
+#define jcl_gm_present_flag			  ( 0x1 << jcl_gm_present)
+#define jcl_peer_present_flag			( 0x1 << jcl_peer_present)
+#define jcl_offset_threshold_flag	    ( 0x1 << jcl_offset_threshold)
+
+/*
+ * jcl_handle_connect
+ *  connects to PTP4L implementation
+ *
```
Comment (AzuraTarmiziIntel) [inline jclklib/proxy/thread.hpp:69]: change it to doxygene style. refer to my earlier commits

File: jclklib/proxy/connect.hpp [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+#include <common/connect_msg.hpp>
+
+namespace JClkLibProxy
+{
+	class Connect
+	{
+	private:
+    public:
+        static int connect();
+    };
```
Comment (yoongsiang2) [inline jclklib/proxy/connect.hpp:11]: line 9-11 need to follow indention.

Comment (yoongsiang2) [reply jclklib/proxy/connect.hpp:11]: btw, what is the reason to create connect() under Connect class?

File: jclklib/proxy/connect.cpp [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        if (epoll_wait( epd, &epd_event, 1, 100) != -1) {
+        //if ( sk->poll(timeout_ms)) {
+            const auto cnt = sk->rcv(buf, bufSize);
+            MNG_PARSE_ERROR_e err = msg.parse(buf, cnt);
+            if(err == MNG_PARSE_ERROR_OK)
+                handle(false);
+        }
+    }
+end_loop:
+    return;
```
Comment (yoongsiang2) [inline jclklib/proxy/connect.cpp:244]: connect.cpp: In function 'void* loop(void*)':
connect.cpp:244:5: error: return-statement with no value, in function returning 'void*' [-fpermissive]
  244 |     return;
      |     ^~~~~~

**Comment by yoongsiang2** [comment]
Added event definition in m4 file to avoid following build error. make command will generate proc.h file accordingly. 

connect.cpp: In function 'bool event_subscription(jcl_handle**)':
connect.cpp:237:16: error: 'NOTIFY_PORT_STATE_NP' was not declared in this scope
  237 |     d.setEvent(NOTIFY_PORT_STATE_NP);
      |                ^~~~~~~~~~~~~~~~~~~~
connect.cpp:238:16: error: 'NOTIFY_ALL' was not declared in this scope
  238 |     d.setEvent(NOTIFY_ALL);
      |                ^~~~~~~~~~

**Review by yoongsiang2 — APPROVED** [review]

**Pull request merged**